cpp/src/interop/to_arrow_device.cu

```diff
@@ -586,6 +378,207 @@ int dispatch_to_arrow_device::operator()<cudf::dictionary32>(cudf::column&& colu
   return NANOARROW_OK;
 }
 
+struct dispatch_to_arrow_device_view {
+  cudf::column_view column;
+  rmm::cuda_stream_view stream;
+  rmm::mr::device_memory_resource* mr;
+
+  template <typename T, CUDF_ENABLE_IF(not is_rep_layout_compatible<T>())>
+  int operator()(ArrowArray*) const
+  {
+    CUDF_FAIL("Unsupported type for to_arrow_device", cudf::data_type_error);
+  }
+
+  template <typename T, CUDF_ENABLE_IF(is_rep_layout_compatible<T>())>
+  int operator()(ArrowArray* out) const
+  {
+    nanoarrow::UniqueArray tmp;
+
+    auto const storage_type = id_to_arrow_storage_type(column.type().id());
+    NANOARROW_RETURN_NOT_OK(initialize_array(tmp.get(), storage_type, column));
+    NANOARROW_RETURN_NOT_OK(set_null_mask(column, tmp.get()));
+    NANOARROW_RETURN_NOT_OK(set_view_to_buffer(column, tmp.get()));
+
+    ArrowArrayMove(tmp.get(), out);
+    return NANOARROW_OK;
+  }
+
+  int set_buffer_view(void const* in_ptr, size_t size, int64_t i, ArrowArray* out) const
+  {
+    ArrowBuffer* buf = ArrowArrayBuffer(out, i);
+    buf->size_bytes  = size;
+    auto const ptr   = reinterpret_cast<uint8_t const*>(in_ptr);
```

Comment:
Can we just do both casts on line 417? This local variable is never used otherwise AFAICT.